nixos/modules/services/web-apps/c2FmZQ-server.nix

```diff
@@ -0,0 +1,138 @@
+{ lib, pkgs, config, ... }:
+
+let
+  inherit (lib) mkEnableOption mkPackageOption mkOption types;
+
+  cfg = config.services.c2FmZQ-server;
+
+  argsFormat = with lib; {
+    type = with types; let
+      valueType = nullOr (oneOf [
+        bool int str
+      ]) // {
+        description = "c2FmZQ-server params";
+      };
+    in valueType;
+    generate = name: value:
+      let mkParam = k: v:
+            if v == null then []
+            else if isBool v then optional v ("--"+k)
+            else [("--"+k) v];
+      in escapeShellArgs (builtins.concatLists (mapAttrsToList mkParam value));
+  };
+in {
+  options.services.c2FmZQ-server = {
+    enable = mkEnableOption "c2FmZQ-server";
+
+    bindIP = mkOption {
+      type = types.str;
+      default = "127.0.0.1";
+      description =  "The local address to use.";
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 8080;
+      description =  "The local port to use.";
+    };
+
+    passphraseFile = mkOption {
+      type = types.str;
+      example = "/run/secrets/c2FmZQ/pwfile";
+      description = "Path to file containing the database passphrase";
+    };
+
+    package = mkPackageOption pkgs "c2FmZQ" { };
+
+    settings = mkOption {
+      type = types.submodule {
+        freeformType = argsFormat.type;
+
+        options = {
+          address = mkOption {
+            internal = true;
+            type = types.str;
+            default = "${cfg.bindIP}:${toString cfg.port}";
+          };
+
+          database = mkOption {
+            type = types.str;
+            default = "%S/c2FmZQ-server/data";
+            description = "Path of the database";
+          };
+
+          verbose = mkOption {
+            type = types.enum [ 1 2 3 ];
+            default = 2;
+            description = "The level of logging verbosity: 1:Error 2:Info 3:Debug";
+          };
+        };
+      };
+      description = ''
+        Configuration for c2FmZQ-server passed as CLI arguments.
+        Run `c2FmZQ-server help` for supported values.
+      '';
+      example = lib.literalExpression ''
+        {
+          verbose = 3;
+          allow-new-accounts = true;
+          auto-approve-new-accounts = true;
+          encrypt-metadata = true;
+          enable-webapp = true;
+        }
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services.c2FmZQ-server = {
+      description = "c2FmZQ is an application that can securely encrypt, store, and share files, including but not limited to pictures and videos.";
+      documentation = [ "https://github.com/c2FmZQ/c2FmZQ" ];
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" "network-online.target" ];
+
+      serviceConfig = {
+        ExecStart = "${cfg.package}/bin/c2FmZQ-server " + argsFormat.generate "c2FmZQ-server" cfg.settings;
```

Comment:
```suggestion
        ExecStart = "${lib.getExe cfg.package} ${argsFormat.generate "c2FmZQ-server" cfg.settings}";
```

Reply:
I believe `mainProgram` is set to `c2FmZQ-client`

Reply:
I did change `mainProgram` to `c2FmZQ-server` already.

Reply:
Isn't the thing with `mainProgram` that you should only set it where it's obvious that a package has a main program? Or in other words, where it's obvious which path you'd get if you use `lib.getExe` on it?

I think this qualifies for not having `mainProgram` set at all.